src/ol/render/canvas/ImageBuilder.js

```diff
@@ -109,12 +110,17 @@ class CanvasImageBuilder extends CanvasBuilder {
   /**
    * @param {import("../../geom/Point.js").default|import("../Feature.js").default} pointGeometry Point geometry.
    * @param {import("../../Feature.js").FeatureLike} feature Feature.
+   * @param {number} [index] Render order index.
    */
-  drawPoint(pointGeometry, feature) {
-    if (!this.image_) {
+  drawPoint(pointGeometry, feature, index) {
+    if (
+      !this.image_ ||
+      (this.maxExtent &&
+        !containsCoordinate(this.maxExtent, pointGeometry.getFlatCoordinates()))
+    ) {
       return;
```

Comment:
We no longer clip point (and multipoint) symbols to the tile extent, to avoid issues with points near tile borders that are not duplicated on neighboring tiles. By clipping to maxExtent, which is set to the tile extent when rendering vector tiles, we avoid duplicated points if they *are* duplicated on neighboring tiles.